### PR TITLE
CERT 7109 Modify Examples to use Address function calls

### PR DESCRIPTION
**Pull request by nivcertora**
https://certora.atlassian.net/browse/CERT-7109
Block by a crash found:
https://certora.atlassian.net/browse/CERT-7280

Ready for review

### CODE REVIEW

File: DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec [hunk 1]
```diff
@@ -202,15 +202,17 @@ rule monotonicityOfMint(uint256 x, uint256 y, address recipient) {
...
     env eM;
     setup(eM);
+    address a;
+    require a == _token0 || a == _token1;
     requireInvariant integrityOfTotalSupply();
     storage init = lastStorage;
     require recipient != currentContract;
     require x > y ;
-    _token0.transfer(eT0, currentContract, x);
+    a.transfer(eT0, currentContract, x);
```
Comment (nd-certora) [inline DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec:211]:
```suggestion
    token.transfer(eT0, currentContract, x);
```

File: DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec [hunk 2]
```diff
@@ -202,15 +202,17 @@ rule monotonicityOfMint(uint256 x, uint256 y, address recipient) {
     env eT0;
     env eM;
     setup(eM);
+    address a;
+    require a == _token0 || a == _token1;
```
Comment (nd-certora) [inline DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec:206]: rename 'a' to  'token' to make it more readable

**Review by nd-certora — APPROVED** [review]

File: DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec (hunk 1 above)
Comment (nivcertora) [reply DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec]: Done

File: DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec (hunk 2 above)
Comment (nivcertora) [reply DEFI/ConstantProductPool/certora/spec/ConstantProductPool.spec]: Done

**Pull request merged**